kong/runloop/balancer/targets.lua

```diff
@@ -0,0 +1,490 @@
+---
+--- manages a cache of targets belonging to an upstream.
+--- each one represents a hostname with a weight,
+--- health status and a list of addresses.
+---
+--- maybe it could eventually be merged into the DAO object?
+---
+
+local singletons = require "kong.singletons"
+
+local dns_client = require "resty.dns.client"
+local upstreams = require "kong.runloop.balancer.upstreams"
+local balancers   -- require at init time to avoid dependency loop
+local dns_utils = require "resty.dns.utils"
+
+local ngx = ngx
+local null = ngx.null
+local ngx_now = ngx.now
+local log = ngx.log
+local string_format = string.format
+local string_match  = string.match
+local ipairs = ipairs
+local tonumber = tonumber
+local table_sort = table.sort
+--local assert = assert
+
+local ERR = ngx.ERR
+local WARN = ngx.WARN
+local DEBUG = ngx.DEBUG
+
+local SRV_0_WEIGHT = 1      -- SRV record with weight 0 should be hit minimally, hence we replace by 1
+local EMPTY = setmetatable({},
+  {__newindex = function() error("The 'EMPTY' table is read-only") end})
+local GLOBAL_QUERY_OPTS = { workspace = null, show_ws_id = true }
+
+
+local targets_M = {}
+
+-- forward local declarations
+local resolve_timer_callback
+local queryDns
+
+function targets_M.init()
+  require "kong.tools.dns"(kong.configuration)    -- configure DNS client
```

Comment:
```suggestion
  require("kong.tools.dns")(kong.configuration)    -- configure DNS client
```

Was a bit hard to read at first 😂